File: 02/012/main.py
def factorsLengthOver(num, maxnum):
    totalFactor = 0
    limit = num
    i = 1
    while i < limit:
        if num % i == 0:
            if num / i == i:
                totalFactor += 1
            else:
                totalFactor += 2
            limit = num / i
            if totalFactor > maxnum:
                return True
        i += 1
    return False

def triangleFactorOver(maxFactor):
    triangle = 0
    i = 1
    while True:
        triangle += i
        if factorsLengthOver(triangle, maxFactor):
            break
        i += 1
    return triangle

print(triangleFactorOver(500))

File: 02/012/main.go
package main

import "fmt"

func FactorsLengthOver(num int, max int) bool {
	var totalFactor = 0
	var limit = num
	for i := 1; i < limit; i++ {
		if num%i == 0 {
			if num/i == i {
				totalFactor += 1
			} else {
				totalFactor += 2
			}
			limit = num / i
			if totalFactor > max {
				return true
			}
		}
	}
	return false
}

func TriangleFactorOver(maxFactor int) int {
	var triangle = 0
	for i := 1; ; i++ {
		triangle += i
		if FactorsLengthOver(triangle, maxFactor) {
			break
		}
	}
	return triangle
}

func main() {
	fmt.Println(TriangleFactorOver(500))
}
